### PR TITLE
Fixes #12752: Datasources plugin scheduler problems after updating/deleting a datasource

### PR DESCRIPTION
**Pull request by fanf**
https://issues.rudder.io/issues/12752

The problem was an non evaluated `RudderIO` on save :/
I've added a test to check for that now.

### CODE REVIEW

File: datasources/src/main/scala/com/normation/plugins/datasources/Repository.scala [hunk 1]
```diff
@@ -168,30 +168,30 @@ class DataSourceRepoImpl(
...
+    private[datasources] val internalRef = Ref.make(Map[DataSourceId, DataSourceScheduler]()).runNow
 
     // utility methods on datasources
     // stop a datasource - must be called when the datasource still in "datasources"
-    private[this] def stop(id: DataSourceId) = {
+    private[datasources] def stop(id: DataSourceId) = {
       DataSourceLoggerPure.debug(s"Stopping data source with id '${id.value}'") *>
-      internalRef.get.map(_.get(id) match {
-        case None      => DataSourceLogger.trace(s"Data source with id ${id.value} was not found running")
+      internalRef.get.flatMap(_.get(id) match {
```
Comment (fanf) [inline datasources/src/main/scala/com/normation/plugins/datasources/Repository.scala:179]: this is the problem. `dss.cancel()` was never evaluated. It dates from rudder 6.0 !

Comment (VinceMacBuche) [reply datasources/src/main/scala/com/normation/plugins/datasources/Repository.scala:179]: It made an IOResult [IOResult[Unit]] ?

Comment (VinceMacBuche) [reply datasources/src/main/scala/com/normation/plugins/datasources/Repository.scala:179]: Maybe we could add return type of stop method

**Comment by fanf** [comment]
PR updated with a new commit

**Review by VinceMacBuche — APPROVED** [review]

**Comment by VinceMacBuche** [comment]
OK, squash merging this PR

**Pull request merged**